db/migrate/20151117005827_create_review_chats.rb

```diff
@@ -0,0 +1,13 @@
+class CreateReviewChats < ActiveRecord::Migration
+  def change
+    create_table :review_chats do |t|
+      t.integer :assignment_id
```

Comment:
The biggest thing I would suggest you change is that for the this migration, you may remove the assignment_id, reviewer_id and team_id. You only need to add a response_map_id and this foreign key points to a record in response_map with type="ReviewResponseMap".
After that:
response_map = ReviewResponseMap.find(review_chat_object.response_map_id)
team_id = response_map.reviewee_id
reviewer_id = response_map .reviewer_id
assignment_id = AssignmentParticipant.find(reviewer_id).parent_id